hack/conformance.sh

```diff
@@ -29,6 +29,10 @@ if [[ ! -z "$SKIP_OUTSIDE_CONN_TESTS" ]]; then
     sonobuoy_args="${sonobuoy_args} --plugin-env kubevirt-conformance.E2E_SKIP=\[outside_connectivity\]"
 fi
 
+if [[ ! -z "$SKIP_ON_ARM64_INFRA" ]]; then
+    sonobuoy_args="${sonobuoy_args} --plugin-env kubevirt-conformance.E2E_SKIP=\[ARM64InfraNotSupported\]"
```

Comment:
I think this was used only for the network connectivity test, which has already a flag that you can use instead. Can you use it?

Having a generic flag that says "arm64 is not supported" is a bit problematic. It is not that the functionality is not there, it usually has to do with other things (e.g. using `kind`).

Reply:
Hi, I thought about to reuse the environmental variable `SKIP_OUTSIDE_CONN_TESTS` or the flag `[outside_connectivity]`, but using `SKIP_OUTSIDE_CONN_TESTS` and `[outside_connectivity]` would skip both IPv4 and IPv6 outside connectivity e2e tests, and I only want to skip IPv6 outside connectivity e2e tests. Here is the original solution and discussions for the issue. https://github.com/kubevirt/kubevirt/pull/10303#discussion_r1341107517

How about change the label's name from "[ARM64InfraNotSupported]" to "[outside_connectivity_ipv6]", which would not lead to misunderstanding? Or do you have any suggestions?

Reply:
- Is the lack of infra support, related to this running using `kind`?
- Ginkgo already has a dedicated label tooling [1], we should use it instead of the inline free text.
- The user facing and logic that I would prefer here, is something like this:
  - User/tool that runs the conformance tests, or the test itself, should identify arm64 and skip the tests that are not supported. I usually prefer the input to be from the one that executes the tests, similar to what was suggested here (but declarative, e.g. `ARCH_ARM64=true make conformance`).
  - The interpretation of running an arm64 execution is this logic: `!(outside_connectivity && IPv6)`. I'm unsure if this requires a REGEX expression or not (maybe the equivalent `!outside_connectivity || !IPv6` will be simpler).
  - Alternatively to the above, depending on the 1st point answer (regarding `kind`), you could add a skip in the test if it runs through `kind` (`SkipIfRunningOnKindInfra` which you created in this PR).

[1] https://onsi.github.io/ginkgo/#spec-labels

Reply:
> Is the lack of infra support, related to this running using `kind`?

I don't think so, as `kind` support dual ipFamily. After setting up dual ipFamily, it seems that both IPv4 and IPv6 e2e testing are works fine within the kind k8s cluster. I think the IPv6 outside connectivity issue may relate to the k8s cluster network configuration (not the one created via `kind` but created via `kubeadm` and deployed on the baremetal server) and the baremetal server network setting on Arm64 platform. I have not solved it yet.

>  The user facing and logic that I would prefer here, is something like this:

Based on your explanation, here is my understanding. Is it correct?
1. We add two labels, `[outside_connectivity]` and `[IPv6]`, to the test case.
2. we can use `--label-filter` to manage the logic, like you said `--label-filter !outside_connectivity || !IPv6`, to skip the test that have bot outside_connectivity and IPv6 labels. This should be done in `tests/conformance/conformance.go`.
3. In the `hack/conformance.sh`, write the code like: 
```
if [[ ! -z "$SKIP_ON_ARM64_INFRA" ]]; then
    sonobuoy_args="${sonobuoy_args} --plugin-env kubevirt-conformance.LABEL_FILTER="\!outside_connectivity \|\| \!IPv6"
fi
```
Thus we can run `SKIP_ON_ARM64_INFRA=true make conformance` and have correct e2e tests to be skipped.

Reply:
> Based on your explanation, here is my understanding. Is it correct?
> 
>     1. We add two labels, `[outside_connectivity]` and `[IPv6]`, to the test case.
> 
>     2. we can use `--label-filter` to manage the logic, like you said `--label-filter !outside_connectivity || !IPv6`, to skip the test that have bot outside_connectivity and IPv6 labels. This should be done in `tests/conformance/conformance.go`.
> 
>     3. In the `hack/conformance.sh`, write the code like:
> 
> 
> ```
> if [[ ! -z "$SKIP_ON_ARM64_INFRA" ]]; then
>     sonobuoy_args="${sonobuoy_args} --plugin-env kubevirt-conformance.LABEL_FILTER="\!outside_connectivity \|\| \!IPv6"
> fi
> ```

Yes, exactly.
I think that we should declare using the env-variable only that we run it for arm64, not that we want to skip something. The skip part is and implementation of running on arm64 infra, as you show above.
i.e. instead of calling it `SKIP_ON_ARM64_INFRA` I would have called it `RUN_ON_ARM64_INFRA`.

---
Thank you for the explanation on the `kind` part.
Generally speaking, I think the connectivity e2e test are problematic, but we never had time to invest in them.